File: app/src/main/java/com/savvasdalkitsis/gameframe/draw/model/RectangleTool.java
package com.savvasdalkitsis.gameframe.draw.model;

import com.savvasdalkitsis.gameframe.grid.model.Grid;

public class RectangleTool extends ScratchTool {

    @Override
    void drawOnScratch(Grid scratch, int startColumn, int startRow, int column, int row, int color) {
        for (int c = Math.min(startColumn, column); c <= Math.max(startColumn, column); c++) {
            scratch.setColor(color, c, startRow);
            scratch.setColor(color, c, row);
        }
        for (int r = Math.min(startRow, row); r <= Math.max(startRow, row); r++) {
            scratch.setColor(color, startColumn, r);
            scratch.setColor(color, column, r);
        }
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/draw/model/RectangleTool.kt
package com.savvasdalkitsis.gameframe.draw.model

import com.savvasdalkitsis.gameframe.grid.model.Grid

class RectangleTool : ScratchTool() {

    override fun drawOnScratch(scratch: Grid, startColumn: Int, startRow: Int, column: Int, row: Int, color: Int) {
        for (c in Math.min(startColumn, column)..Math.max(startColumn, column)) {
            scratch.setColor(color, c, startRow)
            scratch.setColor(color, c, row)
        }
        for (r in Math.min(startRow, row)..Math.max(startRow, row)) {
            scratch.setColor(color, startColumn, r)
            scratch.setColor(color, column, r)
        }
    }
}
